src/convertToObject.js

```diff
@@ -3,16 +3,29 @@
 /**
  * Implement convertToObject function:
  *
- * Function takes a string with styles (see an example in [stylesString.js](./stylesString.js))
+ * Function takes a string with styles (see an example in
+ * [stylesString.js](./stylesString.js))
  * and returns an object where CSS properties are keys
- * and values are the values of related CSS properties (see an exampl in [test file](./convertToObject.test.js))
+ * and values are the values of related CSS properties
+ * (see an exampl in [test file](./convertToObject.test.js))
  *
  * @param {string} sourceString
  *
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const stringToCSS = sourceString
+    .split(';')
+    .map(elem => elem.trim())
+    .filter(elem => elem !== '')
+    .map(elem => elem.split(':').map(property => property.trim()))
+    .reduce((prev, elem) => {
+      return {
+        ...prev, [elem[0]]: elem[1],
+      };
+    }, {});
```

Comment:
Great! But I was advised to use only one iteration instead of several. Try to do the same

Reply:
Can you please explain what you mean? Because I don't really understand how to do it right now😞